### PR TITLE
remove http_signature_test from samples when not supported

### PR DESCRIPTION
**Pull request by tiffmaelite**
For some samples, the openapi.yaml file contains endpoints with http signature authentication. But not all generators support this type of authentication. In some cases (addressed by #15220), this leads to invalid API being generated, because the signature authentication will be incorrectly considered to be a basic or bearer authentication.

The signature scheme was initially added in `petstore-with-fake-endpoints-models-for-testing` by #6058 for jersey2-experimental (now jersey2-java8) only. After the creation of new samples using the same input file, this small "localized" change ended up affecting almost all samples.
Furthermore, this change was not removed when another special yaml file with http-signature-test (#4993) was introduced, and the current PR aims at fixing this omission.

The goal of this PR is to clean up the samples to ensure that they do not contain and show invalid API code.

Once this PR and #15220 are merged, I will be able to submit another PR which explicitly verifies API behavior (e.g. exceptions) in case the openapi.yaml file contains authentication methods not supported by the generator.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Regenerated samples for several modules, so here are mentions of concerned technical committee members for review and approval:
- [ ] c# @mandrean @frankyjuang  @shibayan @Blackclaws @lucamazzanti
- [x] elixir @mrmstn
- [ ] java @bbdouglas @sreeshas  @jfiala  @lukoyanov  @cbornet @jeff9finger @karismann  @Zomzog @lwlee2608
- [ ] javascript @CodeNinjai @frol @cliffano
- [x] k6 @mostafa
- [ ] perl @wing328 @yue9944882
- [ ] php @jebentier , @dkarlovi , @mandrean , @jfastnacht , @ybelenko , @renepardon 
- [ ] ruby @cliffano @zlx @autopp
- [ ] typescript @TiFu  @taxpon  @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce  @akehir  @petejohansonxo @amakhrov @davidgamero @mkusaka
- [ ] dart @jaumard @josh-burton @amondnet @sbu-WBT @kuhnroyal  @agilob @ahmednfwela
- [ ] python @spacether  @krjakbrjak
- [x] C++ @ravinikam @stkrwork @etherealjoy @martindelille @muttleyxd

### CODE REVIEW

**Comment by mostafa** [comment]
LGTM!

**Comment by mrmstn** [comment]
LGTM for elixir

**Comment by mrmstn** [comment]
@tiffmaelite Looks like the CI complains about some compile errors. I'm not quite sure if you have access to the reports but it would be great if you could have a look

I also want to thank you for your contribution and the time you've already invested 😁!

**Comment by tiffmaelite** [comment]
@mrmstn Thank you for the notice. I did not see these yesterday. Of course, I forgot to push the commit in which I removed the non-generated test methods which are no longer relevant.

**Comment by tiffmaelite** [comment]
Note: I aimed this PR at 7.x due to its relation with #15220, but I'd be happy to move it to master if the core team thinks it would be a better target

**Review by muttleyxd — APPROVED** [review]
Changes in C++ code look good to me

File: modules/openapi-generator/src/test/resources/3_0/javascript/petstore-with-fake-endpoints-models-for-testing.yaml
Comment (wing328) [inline modules/openapi-generator/src/test/resources/3_0/javascript/petstore-with-fake-endpoints-models-for-testing.yaml:1277]: @tiffmaelite thanks for the PR but why not still include the test anyway as it shouldn't break the client/server, right?

Comment (wing328) [reply modules/openapi-generator/src/test/resources/3_0/javascript/petstore-with-fake-endpoints-models-for-testing.yaml:1277]: can you please PM me via Slack to further discuss this PR when you've time?

Comment (tiffmaelite) [reply modules/openapi-generator/src/test/resources/3_0/javascript/petstore-with-fake-endpoints-models-for-testing.yaml:1277]: @wing328 will do somewhen tomorrow or early next week.

I also wanted to add a last argument here: what is the point of having files named `petstore-with-fake-endpoints-models-for-testing.yaml` and `petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml` if both contain the http-signature scheme?

But maybe it does not make sense to have both of them side by side, because they are quite hard to maintain (and actually one can see that their content on some endpoints diverges already for no other reason than "someone forgot about the other file")

File: samples/client/petstore/java/apache-httpclient/src/main/java/org/openapitools/client/ApiClient.java
Comment (tiffmaelite) [inline samples/client/petstore/java/apache-httpclient/src/main/java/org/openapitools/client/ApiClient.java:182]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of code

File: samples/client/petstore/java/feign/src/main/java/org/openapitools/client/ApiClient.java
Comment (tiffmaelite) [inline samples/client/petstore/java/feign/src/main/java/org/openapitools/client/ApiClient.java:73]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of code

File: samples/client/petstore/java/okhttp-gson/README.md
Comment (tiffmaelite) [inline samples/client/petstore/java/okhttp-gson/README.md:283]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of documentation

File: samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/ApiClient.java
Comment (tiffmaelite) [inline samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/ApiClient.java:246]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of code

File: samples/client/petstore/java/webclient-jakarta/README.md
Comment (tiffmaelite) [inline samples/client/petstore/java/webclient-jakarta/README.md:252]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of documentation

File: samples/client/petstore/java/webclient-jakarta/src/main/java/org/openapitools/client/ApiClient.java
Comment (tiffmaelite) [inline samples/client/petstore/java/webclient-jakarta/src/main/java/org/openapitools/client/ApiClient.java:155]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of code

File: samples/client/petstore/java/webclient/README.md
Comment (tiffmaelite) [inline samples/client/petstore/java/webclient/README.md:252]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of documentation

File: samples/client/petstore/perl/README.md
Comment (tiffmaelite) [inline samples/client/petstore/perl/README.md:515]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of documentation

File: samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake-json_serializable/doc/FakeApi.md
Comment (tiffmaelite) [inline samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake-json_serializable/doc/FakeApi.md:78]: note for discussion: when http_signature_test is kept, the samples contain this incorrect example code

File: samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/doc/FakeApi.md
Comment (tiffmaelite) [inline samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/doc/FakeApi.md:78]: note for discussion: when http_signature_test is kept, the samples contain this incorrect code example

File: samples/client/petstore/java/apache-httpclient/README.md
Comment (tiffmaelite) [inline samples/client/petstore/java/apache-httpclient/README.md:245]: note for discussion: when http_signature_test is kept, the samples contain this incorrect line of documentation

File: samples/client/petstore/perl/README.md
Comment (tiffmaelite) [reply samples/client/petstore/perl/README.md]: now outdated

**Comment by tiffmaelite** [comment]
This PR is likely to become irrelevant once #15523 and #15520 are merged, because samples will no longer generate invalid code or documentation

File: samples/client/petstore/java/webclient-jakarta/src/main/java/org/openapitools/client/ApiClient.java
Comment (tiffmaelite) [reply samples/client/petstore/java/webclient-jakarta/src/main/java/org/openapitools/client/ApiClient.java:155]: outdated now

**Pull request closed without merging**
